### PR TITLE
print out hydra.searchpath value if available

### PR DESCRIPTION
**Pull request by jieru-hu**
address part of #1469

### hydra.searchpath from primary config
```
$ python /Users/jieru/workspace/hydra-fork/hydra/examples/advanced/config_search_path/my_app.py --info searchpath

Config search path
******************
| Provider                 | Search path                                                                              |
-----------------------------------------------------------------------------------------------------------------------
| hydra                    | pkg://hydra.conf                                                                         |
| main                     | file:///Users/jieru/workspace/hydra-fork/hydra/examples/advanced/config_search_path/conf |
| hydra-colorlog           | pkg://hydra_plugins.hydra_colorlog.conf                                                  |
| hydra-submitit-launcher  | pkg://hydra_plugins.hydra_submitit_launcher.conf                                         |
| hydra.searchpath in main | pkg://additonal_conf                                                                     |
| schema                   | structured://                                                                            |
-----------------------------------------------------------------------------------------------------------------------
```

###  no hydra.searchpath configured
```
(pytest38) jieru-mbp:hydra jieru$ python /Users/jieru/workspace/hydra-fork/hydra/examples/tutorials/basic/running_your_hydra_app/3_working_directory/my_app.py --info searchpath

Config search path
******************
| Provider                | Search path                                                                                                        |
------------------------------------------------------------------------------------------------------------------------------------------------
| hydra                   | pkg://hydra.conf                                                                                                   |
| main                    | file:///Users/jieru/workspace/hydra-fork/hydra/examples/tutorials/basic/running_your_hydra_app/3_working_directory |
| hydra-colorlog          | pkg://hydra_plugins.hydra_colorlog.conf                                                                            |
| hydra-submitit-launcher | pkg://hydra_plugins.hydra_submitit_launcher.conf                                                                   |
| schema                  | structured://                                                                                                      |
------------------------------------------------------------------------------------------------------------------------------------------------
```

###  hydra.searchpath from command line
```
(pytest38) jieru-mbp:hydra jieru$ python /Users/jieru/workspace/hydra-fork/hydra/examples/tutorials/basic/running_your_hydra_app/3_working_directory/my_app.py hydra.searchpath=[file://fake_conf] --info searchpath

Config search path
******************
| Provider                         | Search path                                                                                                        |
---------------------------------------------------------------------------------------------------------------------------------------------------------
| hydra                            | pkg://hydra.conf                                                                                                   |
| main                             | file:///Users/jieru/workspace/hydra-fork/hydra/examples/tutorials/basic/running_your_hydra_app/3_working_directory |
| hydra-colorlog                   | pkg://hydra_plugins.hydra_colorlog.conf                                                                            |
| hydra-submitit-launcher          | pkg://hydra_plugins.hydra_submitit_launcher.conf                                                                   |
| hydra.searchpath in command-line | file://fake_conf                                                                                                   |
| schema                           | structured://                                                                                                      |
---------------------------------------------------------------------------------------------------------------
```

### CODE REVIEW

**Review by omry — CHANGES_REQUESTED** [review]
1. I don't understand why the change here is doing what you are saying it does.
2. Please add tests covering this.

**Comment by omry** [comment]
> > 1. I don't understand why the change here is doing what you are saying it does.
> 
> I've added a comment there. `hydra.searchpath` is processed [here](https://github.com/facebookresearch/hydra/blob/2930a8d1dd5bc6f2ed3dc5c81d0c4558c7a85ed4/hydra/_internal/config_loader_impl.py#L225), and within the function the repo's config sources is updated to include `hydra.searchpath`, therefore it seems to make sense to load the configuration first.

Can you file a bug for this? loading a config should not change global state.

> > 1. Please add tests covering this.
> 
> done!

**Comment by jieru-hu** [comment]
> Can you file a bug for this? loading a config should not change global state.

I'm happy to file an issue, but before doing that...

Loading a config here only changes the [caching repo's state](https://github.com/facebookresearch/hydra/blob/master/hydra/_internal/config_loader_impl.py#L220-L225), does that count as the global state?

**Comment by omry** [comment]
the caching repo is instantiated inside load_configuration so it does not count as global state.

**Comment by jieru-hu** [comment]
> the caching repo is instantiated inside load_configuration so it does not count as global state.

OK, I thought so. No issue is needed then :)

**Review by omry — CHANGES_REQUESTED** [review]
This will probably break now that I fixed the bug.
rebase take advantage of the hydra.runtime.config_sources?

**Comment by jieru-hu** [comment]
the failed tests are due to DeprecationWarnings from nevergrad sweeper, I've created a PR here https://github.com/facebookresearch/hydra/pull/1546

File: hydra/_internal/hydra.py [hunk 1]
```diff
@@ -381,20 +381,29 @@ def _print_search_path(
 
         box: List[List[str]] = [["Provider", "Search path"]]
 
-        for sp in self.config_loader.get_sources():
-            box.append([sp.provider, sp.full_path()])
+        cfg = self._get_cfg(
+            config_name=config_name,
+            overrides=overrides,
+            cfg_type="all",
```
Comment (omry) [inline hydra/_internal/hydra.py:387]:
```suggestion
            cfg_type="hydra",
```

**Review by omry — APPROVED** [review]
overall looks good.
See inline nit.

**Comment by jieru-hu** [comment]
> overall looks good.
> See inline nit.

thanks. fixed!

**Pull request merged**